gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/validation/EndpointGroupsValidationServiceImpl.java

```diff
@@ -42,32 +45,40 @@ public EndpointGroupsValidationServiceImpl(final ConnectorService connectorServi
 
     @Override
     public List<EndpointGroup> validateAndSanitize(List<EndpointGroup> endpointGroups) {
+        // TODO FCY: what if there is no endpoint group ? in v1/2/3, we throw a EndpointMissingException
```

Comment:
Was thinking about that and finally you can delete all the endpoint after creating your api so don't know if that exception make sense.

Reply:
I understand but I'm wondering if it can cause trouble in some process if you don't have any endpoint nor endpoint group.
Maybe a question for @mouligno ?

Reply:
if one or the other are mandatory, the ui should be aligned. From what i remembered only endpointgroup is mandatory.

Reply:
As seen together, we must implement the validation